br/pkg/restore/systable_restore.go

```diff
@@ -158,8 +158,9 @@ func (rc *Client) getDatabaseByName(name string) (*database, bool) {
 		Name:           model.NewCIStr(name),
 		TemporaryName:  utils.TemporaryDBName(name),
 	}
-	for _, t := range schema.Tables {
-		db.ExistingTables[t.Name.L] = t
+	for _, t := range infoSchema.SchemaTables(schema.Name) {
```

Comment:
will it OOM? because infoSchemaV2 is proposed to avoid OOM when table number is large

Reply:
the parameter is only `mysql` or `sys` now, I would change the function name to `getSystemDatabaseByName`.
Besides, it's better to introduce something like iterator in other places.